tests/gh-api-helpers.js

```diff
@@ -0,0 +1,210 @@
+const { Octokit } = require('octokit')
+const axios = require('axios')
+const AdmZip = require('adm-zip')
+
+const ref = process.env.GITHUB_HEAD_REF || process.env.GITHUB_REF
+const owner = 'Kong'
+const repo = 'gateway-test-scheduler'
+const workflow_id = 'integration.yml'
+
+const POLLING_TIMEOUT = 300000
+const RETRY_INTERVAL = 5000
+
+let octokit
+
+const helpers = {
+  setup: () => {
+    octokit = new Octokit({
+      auth: `${process.env.GITHUB_TOKEN}`,
+    })
+  },
+  scheduleAndRun: async (scheduler_res_path, runUUID) => {
+    if (!octokit) {
+      throw new Error('octokit not initialized, call setup() first')
+    }
+
+    try {
+      return await octokit.request(
+        'POST /repos/{owner}/{repo}/actions/workflows/{workflow_id}/dispatches',
+        {
+          owner,
+          repo,
+          workflow_id,
+          ref,
+          inputs: {
+            scheduler_res_path,
+            UUID: runUUID,
+          },
+          headers: {
+            'X-GitHub-Api-Version': '2022-11-28',
+          },
+        },
+      )
+    } catch (error) {
```

Comment:
Would it be better to just propagate the error instead of returning `undefined` implicitly here?